components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java

```diff
@@ -252,10 +251,12 @@ public static void removeCartridge(String cartridgeType) throws RestAPIException
             if (log.isInfoEnabled()) {
                 log.info(String.format("Successfully removed cartridge: [cartridge-type] %s ", cartridgeType));
             }
-        } catch (Exception e) {
-            String msg = "Could not remove cartridge " + e.getLocalizedMessage();
-            log.error(msg, e);
-            throw new RestAPIException(e.getMessage(), e);
+        } catch (RemoteException e) {
+            throw new RestAPIException(e.getMessage());
+        } catch (CloudControllerServiceInvalidCartridgeTypeExceptionException e) {
```

Comment:
We should not catch the specific exception at Util class. Catch it in API class and send the response code accordingly.